src/js/background.js

```diff
@@ -497,7 +474,10 @@ Badger.prototype = {
       chrome.browserAction.setBadgeText({tabId: tabId, text: ""});
       return;
     }
-    var numBlocked = this.blockedTrackerCount(tabId);
+
+    let thisTab = this.tabData[tabId];
+    let numBlocked = thisTab ? thisTab.blockedCount : 0;
```

Comment:
Ah, good points. `updateBadge` is now called in the `tabs.onUpdated` listener, which happens when a tab is updated in some manner. The issue here is that without this call, the `0` won't display on pages that don't register any blocked third parties (e.g. anything on `eff.org`), as we only call `updateCount` when a blocked third party is detected. The `tabs.onUpdated` listener gets called when a new tab is created, and this results in an error in the console, as `this.tabData[tabId]` is still null (and is not created until the `main_frame` request is detected). 

It's a little tricky to solve this; I tried initially getting the badge to display the `0` when we were processing the `main_frame` request in `onBeforeRequest` but it was getting cleared. See comments [here](https://github.com/EFForg/privacybadger/pull/1455#issuecomment-311235502) and [here](https://github.com/EFForg/privacybadger/pull/1455#issuecomment-311387219).

One (untested) idea is to remove the badge counter if, when processing the `main_frame` request, we determine it meets the criteria of (a) PB being disabled or (b) being a background page. 

Any other ideas? 
@ghostwords Thoughts?